rpc/src/lib.rs

```diff
@@ -98,20 +98,55 @@ impl<B, C, SC, P, CT> EthApiT for EthApi<B, C, SC, P, CT> where
 				.map_err(|_| internal_err("fetch runtime chain id failed"))?.into()))
 	}
 
-	fn gas_price(&self) -> BoxFuture<U256> {
-		unimplemented!("gas_price");
+	fn gas_price(&self) -> Result<Option<U256>> {
+		let header = self
+			.select_chain
+			.best_chain()
+			.map_err(|_| internal_err("fetch header failed"))?;
+		Ok(Some(
+			self.client
+				.runtime_api()
+				.gas_price(&BlockId::Hash(header.hash()))
+				.map_err(|_| internal_err("fetch runtime chain id failed"))?
+				.into(),
+		))
 	}
 
 	fn accounts(&self) -> Result<Vec<H160>> {
 		unimplemented!("accounts");
 	}
 
-	fn block_number(&self) -> Result<U256> {
-		unimplemented!("block_number");
+	fn block_number(&self) -> Result<Option<U256>> {
+		let header = self
+			.select_chain
+			.best_chain()
+			.map_err(|_| internal_err("fetch header failed"))?;
+		Ok(Some(
+			self.client
```

Comment:
No need to call runtime API here -- the header above already contains the block number, just return that.